crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1143 @@
+                                    }
+                                    break;
+                                }
+                            }
+                        }
+                        Expr::Seq(SeqExpr {
+                            span: DUMMY_SP,
+                            exprs,
+                        })
+                    }
+                    _ => Expr::Assign(assign_expr),
+                }
+            }
+        }
+    }
+
+    fn replace_update_expr(&mut self, update_expr: UpdateExpr) -> Expr {
+        if !update_expr.prefix {
+            match &*update_expr.arg {
+                Expr::Ident(ident) => {
+                    for (k, v) in self.export_map.iter() {
+                        if ident.to_id() == *k {
+                            let mut expr = Expr::Bin(BinExpr {
+                                span: DUMMY_SP,
+                                op: op!(bin, "+"),
+                                left: Box::new(Expr::Unary(UnaryExpr {
+                                    span: DUMMY_SP,
+                                    op: op!(unary, "+"),
+                                    arg: Box::new(Expr::Ident(ident.clone())),
+                                })),
+                                right: 1.0.into(),
+                            });
+                            for value in v.iter() {
+                                expr = Expr::Call(self.export_call(value.clone(), DUMMY_SP, expr));
+                            }
+                            return Expr::Seq(SeqExpr {
+                                span: DUMMY_SP,
+                                exprs: vec![Box::new(expr), Box::new(Expr::Update(update_expr))],
+                            });
+                        }
+                    }
+                    Expr::Update(update_expr)
+                }
+                _ => Expr::Update(update_expr),
+            }
+        } else {
+            Expr::Update(update_expr)
+        }
+    }
+
+    fn add_export_name(&mut self, key: Id, value: JsWord) {
+        let mut find = false;
+        for (k, v) in self.export_map.iter_mut() {
+            if key == *k {
+                v.push(value.clone());
+                find = true;
+                break;
+            }
+        }
+        if !find {
+            self.export_map.insert(key, vec![value]);
+        }
+    }
+
+    fn add_declare_var_idents(&mut self, ident: &Ident) {
+        self.declare_var_idents.push(ident.clone());
+    }
+
+    fn build_export_call(
+        &mut self,
+        export_names: &mut Vec<JsWord>,
+        export_values: &mut Vec<Expr>,
+    ) -> Vec<Stmt> {
+        match export_names.len() {
+            0 => vec![],
+            1 => vec![self
+                .export_call(export_names.remove(0), DUMMY_SP, export_values.remove(0))
+                .into_stmt()],
+            _ => {
+                let mut props = vec![];
+                while !export_names.is_empty() {
```

Comment:
You can use 

```rust
for (sym, value) in export_names.drain(..).rev().zip(export_values.drain(..).rev()) {

}

```


this will be much more efficient

Reply:
Current code requires `memmove` of all elements per loop, but it's not required.

Reply:
Seems like this issue still exists?